core/objects.go

```diff
@@ -252,6 +265,43 @@ type Challenge struct {
 
 	// Used by dns and dvsni challenges
 	Validation *jose.JsonWebSignature `json:"validation,omitempty"`
+
+	// Contains information about URLs used or redirected to and IPs resolved and
+	// used
+	ValidationRecord []ValidationRecord `json:"validationRecord,omitempty"`
```

Comment:
It seems like having the field name be the same as the type is weird.  Any of these would be better:

```
1. ValidationRecords []ValidationRecord
2. ValidationDetails []ValidationRecord
3. ValidationRecord []ValidationStep
```

My personal preference order is probably 2 ~= 1 > 3